lib/logflare/endpoints.ex

```diff
@@ -126,10 +127,18 @@ defmodule Logflare.Endpoints do
     iex> parse_query_string("select @testing from date")
     {:ok, %{parameters: ["testing"]}}
   """
-  @spec parse_query_string(String.t()) :: {:ok, %{parameters: [String.t()]}} | {:error, any()}
-  def parse_query_string(query_string) do
-    with {:ok, declared_params} <- Logflare.Sql.parameters(query_string) do
-      {:ok, %{parameters: declared_params}}
+  @spec parse_query_string(:bq_sql | :pg_sql, String.t(), [Query.t()], [Alert.t()]) ::
+          {:ok, %{parameters: [String.t()], expanded_query: String.t()}} | {:error, any()}
+  def parse_query_string(language, query_string, endpoints \\ [], alerts \\ [])
```

Comment:
Instead of multiple default arguments, you can also consider a single list of options:

```suggestion
  def parse_query_string(language, query_string, opts \\ [])
    opts = Keyword.validate!(opts, [endpoints: [], alerts: []])
```

Or, if you expect both to be generally passed, you can skip the default `\\ []` values, so the call is always explicit.

It mostly depends on how you expect to be called most of the time, and if it's likely to have more options in the future.

Reply:
Removed the default values as it should always be explicit in the foreseeable future

Reply:
thanks for the recommendation for the Keyword.validate!/2, it'll be definitely something to adopt once this function becomes more and more complex